### ISSUE TITLE
Upgrade to ESLint 8

### ISSUE THREAD

**Issue by fasttime** [adametry/gulp-eslint#250]
[~~ESLint 7~~](https://github.com/eslint/eslint/releases/tag/v7.0.0) [ESLint 8](https://github.com/eslint/eslint/releases/tag/v8.0.0) has been released and it includes some important breaking changes. Any plans to support the new version?

**Comment by kuyezhiying** [comment]
+1 for upgrade to eslint v7.0.0, here is a real case, I am recently working on migration from tslint to eslint, and I found eslint has an important fix on the id-blacklist rule in v7, but unfortunately gulp-eslint doesn't support the newest version and I got many complaints in the code, unlike gulp-tslint, seems the API for gulp-eslint cannot specify a version for the dependent eslint, so I have to wait for a gulp-eslint package update to support eslint 7.

**Comment by tmartin2089** [comment]
Same, gulp-eslint is broken on eslint v7.0, ran into this converting a repo to TS

Running eslint directly from .bin works just fine, but gulp-eslint is unable to find my config files in 7.0

**Comment by silviuburceadev** [comment]
@adametry @shinnn do you need help maintaining this project? We could help!

**Comment by silviuburceadev** [comment]
ESLint 7.3.0 was released on Jun 19th, is it time to think about forking this project to leverage ESLint 7.x features in gulp?

**Comment by fasttime** [comment]
If it helps, I have created [a fork](https://github.com/fasttime/gulp-eslint-new) that uses ESLint 8 for linting. It already supports all [new options](https://eslint.org/docs/developer-guide/nodejs-api#%E2%97%86-new-eslint-options), but also still accepts options in the old gulp-elint/CLIEngine format for compatibility. There is still work to be done: make some adjustments for a couple of things that have changed in new versions of ESLint, improve the output and update the documentation.

**Comment by audishos** [comment]
With yarn, I was able to work-around this by adding "eslint" to "resolutions" in my `package.json`.

```
{
  // ...
  "resolutions": {
    "eslint": "^7.8.1"
  }
}
```

I believe that you can do something similar with `npm-shrinkwrap.json` if you use npm.

**Comment by leepowelldev** [comment]
Really eslint should be a peer dependency

**Comment by paulshryock** [comment]
Any progress on this?

**Comment by Charles-Gerber** [comment]
> With yarn, I was able to work-around this by adding "eslint" to "resolutions" in my `package.json`.
> 
> ```
> {
>   // ...
>   "resolutions": {
>     "eslint": "^7.8.1"
>   }
> }
> ```
> 
> I believe that you can do something similar with `npm-shrinkwrap.json` if you use npm.

Yes it seems to be working for me with npm-shrinkwrap.json (need to fix all the the new eslint errors I have in my code now).

Any news on a coming fix?

**Comment by danny007in** [comment]
Make this plugins also like https://github.com/olegskl/gulp-stylelint
because no needed to upgrade eslint inside this project frequently.

**Comment by morganney** [comment]
You can always create your own Gulp task called `lint` using the [Node.js API from ESLint](https://eslint.org/docs/developer-guide/nodejs-api#eslint-class). Then you can use whatever version of ESLint you want with Gulp.

Here's a start:

```js
import { ESLint } from 'eslint'
import { argv } from 'yargs'
import log from 'fancy-log'
import colors from 'ansi-colors'
import PluginError from 'plugin-error'

const fileGlobs = ['src/**/*.js']
async function lint() {
  const { fix } = argv

  try {
    const eslint = new ESLint({ fix })
    const formatter = await eslint.loadFormatter('stylish')
    const results = await eslint.lintFiles(fileGlobs)
    const errors = ESLint.getErrorResults(results)
    const resultText = formatter.format(results)

    if (fix) {
      await ESLint.outputFixes(results)
    }

    if (errors.length) {
      log(resultText)
      // Beep sound in console
      process.stdout.write('\u0007')
    } else {
      log(colors.green('✓ Linting Successful'))
    }
  } catch (err) {
    throw new PluginError('lint', err)
  }
}

export { lint }
```

**Comment by fasttime** [comment]
I finished the remaining work on [gulp-eslint-new](https://github.com/fasttime/gulp-eslint-new), so I thought I would provide an update here. ESLint 8 and all related features should be working as expected now, as described in the documentation and in the updated examples. If anybody decides to try it out and give me a feedback, that would be greatly appreciated. Contributions would be even more welcome of course.

**Comment by jimmywarting** [comment]
Sad to see packages being split apart cuz one is no longer being updated...

**Comment by ronilaukkarinen** [comment]
@fasttime Thanks for this! Works great!
